system/lib/wasm_worker/library_wasm_worker.c

```diff
@@ -37,20 +40,24 @@ emscripten_wasm_worker_t emscripten_create_wasm_worker(void *stackLowestAddress,
 	assert(emscripten_stack_get_base() <= (uintptr_t)stackLowestAddress || emscripten_stack_get_end() >= (uintptr_t)stackLowestAddress + stackSize
 		&& "When creating a Wasm Worker, its stack should be located either in global data or on the heap, not on the calling thread's own stack!");
 
-	// The Worker's TLS area will be spliced off from the stack region.
-    // We expect TLS area to need to be at most 16 bytes aligned
+	// We expect TLS area to need to be at most 16 bytes aligned
 	assert(__builtin_wasm_tls_align() == 0 || 16 % __builtin_wasm_tls_align() == 0);
 
-#ifndef NDEBUG
-	uint32_t tlsSize = (__builtin_wasm_tls_size() + 15) & -16;
-	assert(stackSize > tlsSize);
-#endif
+	// The Worker's TLS area will be spliced off from the stack region, so the
+	// stack needs to be at least as large as the TLS region.
+	uint32_t tlsSize = ROUND_UP(__builtin_wasm_tls_size(), 16);
+	if (stackSize <= tlsSize) {
+		return 0;
+	}
```

Comment:
This change does not seem that useful? It regresses code size, and silently returning zero here on failure changes the semantics that callers might not be expecting that their worker is never going to start. I am not sure what kind of codebase would find this return zero helpful?

Reply:
OK we can just assert.

This line of the docs implies that these can fail with 0 though: https://github.com/emscripten-core/emscripten/blob/347262aec9c4450e34b6af617d1420dbda2f6662/system/include/emscripten/wasm_worker.h#LL25C28-L25C28